### PR TITLE
Remove Vert.x Axle/RX managed instances

### PR DESCRIPTION
**Pull request by cescoffier**
These features have been deprecated in February 2020.
It also simplifies the Vert.x 4 migration, as it reduces the API surface.

### CODE REVIEW

**Comment by gsmet** [comment]
A big +1 for this but apparently there are some remaining bits:
```
Error:  Failures: 
Error:    ReactiveStreamsOperatorsITCase>ReactiveStreamsOperatorsTestCase.testRxJava2:22 1 expectation failed.
Response body doesn't match expectation.
Expected: is "DEF"
  Actual: RESTEASY003210: Could not find resource for full path: http://localhost:8081/reactive/stream-rx
```

**Comment by cescoffier** [comment]
Yeah, I didn't run all the tests on my machine (I have another heating system). I will fix all the remaining issues.

**Comment by geoand** [comment]
💯

**Comment by cescoffier** [comment]
@gsmet @geoand here we go, the last tests have been fixed.

**Review by geoand — APPROVED** [review]
👍🏼

**Review by gsmet — APPROVED** [review]
Very nice cleanup, thanks!

**Comment by gsmet** [comment]
@mkouba we let you have a look as @cescoffier has requested your review :).

File: build-parent/pom.xml [hunk 1]
```diff
@@ -37,7 +37,6 @@
         <graal-sdk.version-for-documentation>20.3.1</graal-sdk.version-for-documentation>
         <mandrel.version-for-documentation>20.3</mandrel.version-for-documentation>
         <rest-assured.version>4.3.2</rest-assured.version>
-        <axle-client.version>1.1.0</axle-client.version>
         <mutiny-client.version>1.1.0</mutiny-client.version>
```
Comment (famod) [inline build-parent/pom.xml:40]: This property doesn't seem to be referenced anymore? Or is there some black magic in the `docs` module?

**Review by mkouba — APPROVED** [review]
Looks good!

File: build-parent/pom.xml (hunk 1 above)
Comment (famod) [reply build-parent/pom.xml:40]: @cescoffier Maybe you want to have a look at this ^ even though the PR has been merged already.

File: docs/src/main/asciidoc/mailer.adoc [hunk 2]
```diff
@@ -411,8 +411,6 @@ If you need fine control on how the mail is sent, for instance if you need to re
 Three API flavors are exposed:
```
Comment (gastaldi) [inline docs/src/main/asciidoc/mailer.adoc:411]: Looks like it's no longer Three?

**Comment by rsvoboda** [comment]
@cescoffier https://github.com/quarkusio/quarkus/wiki/Migration-Guide-1.12 should have entry about this removal

**Comment by cescoffier** [comment]
@rsvoboda done!

**Comment by rsvoboda** [comment]
@cescoffier thanks, maybe point out that Mutiny APIs should be use? To indicate migration path.

**Pull request merged**